### PR TITLE
slack: Add validate_certs option in plugin

### LINKED ISSUE #45682: slack callback plugin doesn't expose options for certificates
##### SUMMARY

I'm using the [`slack` callback plugin](https://docs.ansible.com/ansible/latest/plugins/callback/slack.html) to work around the lack of [Mattermost](https://mattermost.org) support, as Mattermost claims it has a Slack-compatible API.  When I attempt to use this plugin, I get an error that it is unable to verify the certificates to the Webhook URL I give it.  There are possibly two issues here:

1. The callback plugin isn't making proper use of the certificate bundles that have this custom cert added to the trust store
2. There are not options to configure how to use the plugin with a custom 3rd party CA bundle, **or** a way to set the `validate_certs` option to `False`.  This is explained in the error, but is not exposed to the user for configuration.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

slack plugin

##### ANSIBLE VERSION

```paste below
ansible 2.6.4
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/klauer/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
```

##### CONFIGURATION

```paste below
ANSIBLE_SSH_ARGS(/mnt/c/Users/266891/dev/eams-ansible/meam/ansible.cfg) = -C -o ControlMaster=auto -o ControlPersist=300s
CACHE_PLUGIN(/mnt/c/Users/266891/dev/eams-ansible/meam/ansible.cfg) = jsonfile
CACHE_PLUGIN_CONNECTION(/mnt/c/Users/266891/dev/eams-ansible/meam/ansible.cfg) = ~/.ansible/facts.d/
DEFAULT_CALLBACK_WHITELIST(/mnt/c/Users/266891/dev/eams-ansible/meam/ansible.cfg) = ['timer', 'slack', 'yaml']
DEFAULT_FORKS(/mnt/c/Users/266891/dev/eams-ansible/meam/ansible.cfg) = 5
DEFAULT_HOST_LIST(/mnt/c/Users/266891/dev/eams-ansible/meam/ansible.cfg) = [u'/mnt/c/Users/266891/dev/eams-ansible/meam/inventory/meam-inventory.ini']
DEFAULT_LOG_PATH(/mnt/c/Users/266891/dev/eams-ansible/meam/ansible.cfg) = /home/klauer/.ansible/ansible.log
DEFAULT_MODULE_PATH(/mnt/c/Users/266891/dev/eams-ansible/meam/ansible.cfg) = [u'/mnt/c/Users/266891/dev/eams-ansible/meam/library']
DEFAULT_ROLES_PATH(/mnt/c/Users/266891/dev/eams-ansible/meam/ansible.cfg) = [u'/mnt/c/Users/266891/dev/eams-ansible/meam/roles']
DEFAULT_VAULT_PASSWORD_FILE(/mnt/c/Users/266891/dev/eams-ansible/meam/ansible.cfg) = /home/klauer/ansible_vault_password.txt
HOST_KEY_CHECKING(/mnt/c/Users/266891/dev/eams-ansible/meam/ansible.cfg) = False
```

##### OS / ENVIRONMENT

Running Ansible from Ubuntu 16.04.5 LTS on Windows 10 using WSL.

##### STEPS TO REPRODUCE

```yaml
 [WARNING]: Could not submit message to Slack: Failed to validate the SSL certificate for mattermosttest.server.com:443. Make sure your managed systems have a valid CA certificate installed. You can use validate_certs=False if
you do not need to confirm the servers identity but this is unsafe and not recommended. Paths checked for this platform: /etc/ssl/certs, /etc/pki/ca-trust/extracted/pem, /etc/pki/tls/certs, /usr/share/ca-certificates/cacert.org,
/etc/ansible. The exception msg was: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590).
```

Note, that this certificate **is** actually in the trust store, as I can `curl https://mattermosttest.server.com:443` and get a response back, which I was unable to do **before** putting the certificate in my trust store.

##### ACTUAL RESULTS

```paste below
 [WARNING]: Could not submit message to Slack: Failed to validate the SSL certificate for mattermosttest.server.com:443. Make sure your managed systems have a valid CA certificate installed. You can use validate_certs=False if
you do not need to confirm the servers identity but this is unsafe and not recommended. Paths checked for this platform: /etc/ssl/certs, /etc/pki/ca-trust/extracted/pem, /etc/pki/tls/certs, /usr/share/ca-certificates/cacert.org,
/etc/ansible. The exception msg was: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590).
```

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
This fix adds a configurable parameter for slack plugin.

Fixes: #45682

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lib/ansible/plugins/callback/slack.py

##### ANSIBLE VERSION

```paste below
2.8-devel
```

### CODE REVIEW

File: lib/ansible/plugins/callback/slack.py [hunk 1]
```diff
@@ -42,6 +42,15 @@
         ini:
           - section: callback_slack
             key: username
+      validate_certs:
+        description: validate the SSL certificate of the Slack server. (For HTTPS URLs)
+        env:
+          - name: SLACK_VALIDATE_CERT
```
Comment (sivel) [inline lib/ansible/plugins/callback/slack.py]: This env name should probably be plural, to match the option and ini key.

Comment (sivel) [reply lib/ansible/plugins/callback/slack.py]: `SLACK_VALIDATE_CERTS`

**Comment by Akasurde** [comment]
@sivel Could you please take another look on new changes ? Thanks.

**Review by sivel — APPROVED** [review]

File: lib/ansible/plugins/callback/slack.py [hunk 2]
```diff
@@ -42,6 +42,15 @@
         ini:
           - section: callback_slack
             key: username
+      validate_certs:
+        description: validate the SSL certificate of the Slack server. (For HTTPS URLs)
```
Comment (sivel) [inline lib/ansible/plugins/callback/slack.py:46]: Please add a `version_added` to this as well.

**Review by sivel — CHANGES_REQUESTED** [review]

**Comment by Akasurde** [comment]
@sivel Could you please take another look on new changes ? Thanks.

**Review by sivel — APPROVED** [review]

**Comment by sivel** [comment]
@Akasurde sorry for the delay.  I don't see when people make commits, only comments.

This is merged now.

**Comment by Akasurde** [comment]
@sivel no problem. Thanks for review and merge.

**Pull request merged**